File: hw/sanchez/u3/HW20TemplateMethod/TemplatePY/ec/edu/espe/cafe/model/Coffee.py
from model.CaffeineBeverage import CaffeineBeverage

class Coffee(CaffeineBeverage):
    def brew(self):
        print("Dripping Coffee through filter")

    def addCondiments(self):
        print("Adding Sugar and Milk")

    def wantsCondiments(self):
        answer = self.getUserInput()
        return answer is not None and answer.lower().startswith("y")

    def getUserInput(self):
        try:
            answer = input("Would you like milk an dsugar with your coffee (y/n)? ")
            return answer
        except Exception:
            print("Error reading input.")
            return None

File: hw/suarez/U3/HW21TemplateBeverages/src/main/java/ec/edu/espe/cafe/model/Coffee.java
package ec.edu.espe.cafe.model;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * @author bernardo
 */

public class Coffee extends CaffeineBeverage {
    
    @Override
    public void brew(){
        System.out.println("Dripping coffee through filter\n");
    }
    
    @Override
    public void addCondiments(){
        System.out.println("Adding sugar and milk");
    }
    
    @Override
    public boolean wantsCondiments(){
        String answer = getUserInput();
        
        if(answer.toLowerCase().startsWith ("y")){
            return true;
        } else {
            return false;
        }
    }
    
    private String getUserInput(){
        String answer = null;
    System.out.println("Would you like sugar and milk with your coffee (y/n)?");

    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    
    try {
        answer = in.readLine();
    } catch (IOException e) {
        System.err.println("IO error trying to read your answer");
    }

    return answer;
    }
}